### PR TITLE
Gcp 14 remove provider context

### PR DESCRIPTION
**Pull request by funkyHat**
JIRA: GCP-14

### CODE REVIEW

**Comment by funkyHat** [comment]
Travis seems broken (unable to `git fetch origin +refs/pull/20/merge`)

File: plugin.yaml [hunk 1]
```diff
@@ -57,15 +48,52 @@ node_types:
           in default zone.
         type: string
         default: ''
+      tags:
```
Comment (idanmo) [inline plugin.yaml]: In general, please add descriptions to all new properties/inputs.

File: cloudify_gcp/constants.py [hunk 2]
```diff
@@ -61,7 +60,28 @@
 
 GCP_OP_DONE = 'DONE'
 
-MANAGER_HOMEDIR = os.path.join(os.sep, 'tmp', 'home')
-GCP_DEFAULT_CONFIG_PATH = os.path.join(MANAGER_HOMEDIR, 'gcp_config.json')
+MANAGER_PLUGIN_FILES = os.path.join('/etc', 'cloudify', 'gcp_plugin')
```
Comment (funkyHat) [inline cloudify_gcp/constants.py]: Default `gcp_config` dict as a file (i.e. project, auth dict/filepath, default network, zone)

File: cloudify_gcp/compute/firewall.py [hunk 3]
```diff
@@ -126,78 +144,67 @@ def list(self):
...
-    ctx.instance.runtime_properties.pop(constants.SOURCE_TAGS, None)
-    delete(**kwargs)
+def delete(**kwargs):
+    gcp_config = utils.get_gcp_config()
+    firewall_name = ctx.instance.runtime_properties.get('name', None)
+    if firewall_name:
+        network = utils.get_network(ctx)
+        firewall = FirewallRule(gcp_config,
+                                ctx.logger,
+                                name=ctx.instance.runtime_properties['name'],
```
Comment (funkyHat) [inline cloudify_gcp/compute/firewall.py]: Yes :)

File: cloudify_gcp/compute/instance.py [hunk 4]
```diff
@@ -277,136 +304,191 @@ def get_instance_ssh_keys():
...
         if not ip_address:
             raise GCPError('{} is set, but ip_address is not set'
                            .format(constants.USE_EXTERNAL_RESOURCE))
-        instance.add_access_config(ip_address)
-    else:
-        instance.add_access_config()
+    elif ip_node.type == 'cloudify.gcp.nodes.Address':
+        ip_address = ctx.target.instance.runtime_properties['address']
+
+    instance.add_access_config(ip_address)
```
Comment (funkyHat) [inline cloudify_gcp/compute/instance.py]: Good point. We should be raising a `NonRecoverableError` if we get here and `ip_address` is still `''` (will probably fix with an `else:` above)

File: cloudify_gcp/compute/instance.py [hunk 5]
```diff
@@ -415,73 +497,114 @@ def add_external_ip(instance_name, **kwargs):
...
-    instance.tags.extend(
-        provider_config[constants.AGENTS_SECURITY_GROUP]
-        .get(constants.SOURCE_TAGS))
+        ctx.operation.retry(
+                'The instance has not yet created network interface', 10)
+
+
+def get_ssh_keys():
+    instance_keys = ctx.instance.runtime_properties.get(constants.SSH_KEYS, [])
+    install = ctx.node.properties['install_agent']
```
Comment (funkyHat) [inline cloudify_gcp/compute/instance.py]: Fixed just below (`if not any([...])`)

File: plugin.yaml (hunk 1 above)
Comment (funkyHat) [inline plugin.yaml]: All properties now have descriptions

**Pull request merged**